var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -25,47 +25,152 @@ class Lammps(CMakePackage, CudaPackage):
 
     maintainers = ["rbberger"]
 
+    # rules for new versions and deprecation
+    # * new stable versions should be marked as preferred=True
+    # * a stable version that has updates and any of its outdated update releases should be
+    #   marked deprecated=True
+    # * only the latest update of any given stable release should be marked preferred=True
+    # * patch releases older than a stable release should be marked deprecated=True
     version("develop", branch="develop")
+    version("20221222", sha256="75372ee7ef982767fc4ed4dc95e20ddca8247419adeb0c1276c40e43d1eab955")
     version("20221103", sha256="d28517b84b157d4e46a1a64ed787b4662d8f2f5ade3f5a04bb0caed068f32f7e")
     version("20220915", sha256="392b8d35fc7919b0efaf8b389259a9b795a817e58e75e380467c63d03a0dab75")
     version("20220803", sha256="f37cb0b35c1682ffceae5826aadce47278aa7003099a1655fcea43acd7d37926")
     version(
-        "20220623.2", sha256="fdb5474135b17005030cd78ac85762fecf0944bdd27a81fbbe79b2b4522ccae7"
+        "20220623.2",
+        sha256="8a560213e83919623525c4a7c4b5f0eda35cdf3b0c0e6548fd891379e04ca9e6",
+        preferred=True,
     )
     version(
-        "20220623.1", sha256="1d4fb06a4621b271e8b54f31fef241d0569717a69c1b7646a2b76fe5ce47c966"
+        "20220623.1",
+        sha256="58e3b2b984f8935bb0db5631e143be2826c45ffd48844f7c394f36624a3e17a2",
```

Comment:
On a second check, after fetching the PR, `spack checksum` indicates the sha256 should not be change, i.e.,:

```suggestion
        sha256="1d4fb06a4621b271e8b54f31fef241d0569717a69c1b7646a2b76fe5ce47c966",
```

Reply:
I've fixed my logical mistake, added some necessary `preferred=True`, and incorporated the changes from #35133. Could you please recheck? I went through all stable releases (from 2020 upward) and ran `spack checksum` on them. I would be surprised if there is still some mismatch (unless there are some proxy troubles on my end). Can you verify that they pull the `stable_` tarballs as intended?